dali/operators/generic/slice/slice_attr.h

```diff
@@ -48,47 +56,74 @@ class SliceAttr {
       axes_ = spec.GetRepeatedArgument<int>("axes");
       axis_names_ = TensorLayout{};
     }
+    has_start_ = start_.IsDefined() || rel_start_.IsDefined();
+    has_end_ = end_.IsDefined() || rel_end_.IsDefined();
+    has_shape_ = shape_.IsDefined() || rel_shape_.IsDefined();
+
+    DALI_ENFORCE(!(has_end_ && has_shape_),
+        "``end``/``rel_end`` can't be provided together with ``shape``/``rel_shape``.");
   }
 
   template <typename Backend>
   void ProcessArguments(const workspace_t<Backend> &ws) {
-    DALI_ENFORCE(ws.NumInput() == 3,
-      "Expected 3 inputs. Received: " + std::to_string(ws.NumInput()));
-    // slice args are CPU inputs
     auto curr_batch_size = ws.GetInputBatchSize(0);
-    const auto &crop_anchor = ws.template InputRef<CPUBackend>(1);
-    const auto &crop_shape = ws.template InputRef<CPUBackend>(2);
-    DALI_ENFORCE(crop_anchor.type().id() == crop_shape.type().id(),
-                 make_string("Anchor and shape should have the same type. Got: ",
-                             crop_anchor.type().id(), " and ", crop_shape.type().id()));
-    auto args_dtype = crop_anchor.type().id();
-    TYPE_SWITCH(args_dtype, type2id, ArgsType, SLICE_ARGS_TYPES, (
-      auto anchor_view = view<const ArgsType>(crop_anchor);
-      auto shape_view = view<const ArgsType>(crop_shape);
+    int ndim = ws.GetInputDim(0);
+
+    auto args_shape = TensorShape<1>(ndim);
+    if (!axes_.empty() || !axis_names_.empty()) {
+      args_shape[0] = std::max(static_cast<int>(axes_.size()),
+                               static_cast<int>(axis_names_.size()));
+    }
+
+    if (start_.IsDefined())
+      start_.Acquire(spec_, ws, curr_batch_size, args_shape);
+    else if (rel_start_.IsDefined())
+      rel_start_.Acquire(spec_, ws, curr_batch_size, args_shape);
+
+    if (end_.IsDefined())
+      end_.Acquire(spec_, ws, curr_batch_size, args_shape);
+    else if (rel_end_.IsDefined())
+      rel_end_.Acquire(spec_, ws, curr_batch_size, args_shape);
+    else if (shape_.IsDefined())
+      shape_.Acquire(spec_, ws, curr_batch_size, args_shape);
+    else if (rel_shape_.IsDefined())
+      rel_shape_.Acquire(spec_, ws, curr_batch_size, args_shape);
+
+    if (has_end_ || has_shape_) {  // ``start`` can be omitted
+      DALI_ENFORCE(!spec_.HasArgument("normalized_anchor")
+                && !spec_.HasArgument("normalized_shape"),
+        "``normalized_anchor``/``normalized_shape`` is only relevant when using positional "
```

Comment:
The fact that they are irrelevant doesn't mean they are forbidden (could be ignored, for example).
```suggestion
        "``normalized_anchor`` and ``normalized_shape`` can be used only with positional "
```